lib/rspec/expectations/failure_aggregator.rb

```diff
@@ -0,0 +1,159 @@
+module RSpec
+  module Expectations
+    # @private
+    class FailureAggregator
+      attr_reader :block_label, :metadata
+
+      def aggregate
+        RSpec::Support.with_failure_notifier(self) do
+          begin
+            yield
+          rescue ExpectationNotMetError => e
+            failures << e
+          rescue Exception => e
```

Comment:
I would expect this to be flagged by rubocop. Should comment why explicit `Exception` is the right class here, since it feels pretty "deep" in the stack.

If I have N long running aggregated failures, am I still able to ctrl+c effectively?